Mathlib/Analysis/Convex/Piecewise.lean

```diff
@@ -40,14 +40,14 @@ theorem convexOn_univ_piecewise_Iic_of_antitoneOn_Iic_monotoneOn_Ici
     (h_anti : AntitoneOn f (Set.Iic e)) (h_mono : MonotoneOn g (Set.Ici e)) (h_eq : f e = g e) :
     ConvexOn 𝕜 Set.univ ((Set.Iic e).piecewise f g) := by
   refine ⟨convex_univ, fun x _ y _ a b ha hb hab ↦ ?_⟩
-  by_cases hx : x ≤ e <;> by_cases hy : y ≤ e <;> push_neg at hx hy
+  obtain hx | hx := le_or_gt x e <;> obtain hy | hy := le_or_gt y e
```

Comment:
The changed `push_neg` behaviour in this file is because the old implementation was affected by the issue https://github.com/leanprover/lean4/issues/9893. The new implementation is not affected by that issue.  I'll add a tests to `push_neg` and `ring_nf` documenting the behaviour.